### ISSUE TITLE
Update to the Polkadot RE Spec should trigger update to the copy in the research repo

### ISSUE THREAD

**Issue by drskalman** [w3f/polkadot-spec#28]
The CI should clone the research repo copy the PDF from

/polkadot-spec/runtime-environment-spec/polkadot_re_spec.pdf

to 
    /w3f-research/docs/pdf/polkadot_re_spec.pdf

and commit but it needs to have permission somehow which I don't how. Alternatively it can be installed 
locally as github filter or something.

**Comment by FlorianFranzen** [comment]
We should discuss this further in #158.

**Comment by drskalman** [comment]
- CI for making release, containing host spec, runtime spec.
   - CI commits to research repo? maybe discuss it with @FatemeShirazi/@keorn.
   - Diffing between major releases maybe indicated by commit message but we'll make a separete issue for this.

**Comment by FlorianFranzen** [comment]
I tried to get TeXmacs to create diffs on the command line and even though it should in theory be quite simple, I have not got it working yet.

To create the diff we need load the current tree and the tree of the file at a certain version, pass them through `(compare-version t1 t2)` and then render the result. 
Luckily we are provided with `(compare-with-older url)` (and _...-with-newer_), which does all that for you, as long as the current version is the current buffer. However, just calling texmacs with the file does not work, as the `(current-buffer)` is not set at the time command supplied with `-x` are executed. So you are forced to use `-c`, which converts the file first, but at least allows us to use `(compare-with-older url)` in a `-x` flag.
The url for the compare function is `tmfs` encoded. To refer to the version of a file at a certain revision, just use the following scheme (see `(version-revision-url head rev)` for implementation):
```man
tmfs://revision/<git-revision-hash>/file/<file-path>"
```
Now all of this can be combined to the following command, however at the moment this still fails, because the _tmfs_ link is not recognized. However, when the command is run inside TeXmacs, it produces the expected result. 
```shell
texmacs -c input.tm output.pdf  -x "(compare-with-older \"tmfs://revision/$GITHUB_BASE_REF/file/path/to/input.tm\")" [-q]
```

**Comment by FlorianFranzen** [comment]
The `-c` flag just adds `(load-buffer <input> :strict)` and `(export-buffer <output>)` to the list of commands, so we can just do the same, but in the right order.

**Comment by FlorianFranzen** [comment]
I opened a pull-request with an initial version in PR #249.

@drskalman: It would be great if you could take a look.

**Comment by drskalman** [comment]
there is problem with the diffing included subparts. @FlorianFranzen will spend sometime to fix it by fixing the scheme code.

**Comment by drskalman** [comment]
we should choose a good release of host spec to choose as version 0.0.1

**Comment by lamafab** [comment]
@lamafab @drskalman review this and approve PR -> https://github.com/w3f/polkadot-spec/pull/249.

**Comment by lamafab** [comment]
@FlorianFranzen discuss with @infinity0  -> Release for Research PDFs (hosting, releases, etc.)

**Comment by FlorianFranzen** [comment]
@infinity0 We publish the latest specification on GitHub releases. The actual PDFs are build and uploaded once you trigger a release. You can access the latest version [here](https://github.com/w3f/polkadot-spec/releases/latest). How would you like to have these files integrated with the research page?

**Comment by infinity0** [comment]
The spec is in a separate repo so the website could link or embed these release PDFs directly, this seems fine.

What we don't have a good process for, is for papers also hosted in the research repo. I'll continue commenting on w3f/research-internal#684

**Comment by infinity0** [comment]
Is there a copy of the spec in the research repo? If so, we should delete it, and link to the release PDF directly.

It also looks like github configure their webserver so you are forced to download the PDF instead of viewing it inline. So the release process for this spec should also upload it to some part of the research website (ideally with the same hostname), then we can link to it from the other parts of the website.

**Comment by lamafab** [comment]
@FlorianFranzen will update this.

**Comment by drskalman** [comment]
@FlorianFranzen noted that the artifacts are not permanent. We are going to put it on a github page with the same consistant name which can be browsed using research github website.

**Comment by FlorianFranzen** [comment]
The final step in archiving this was implemented in w3f/research#114.
